src/Sensor_TEST.cc

```diff
@@ -349,6 +349,61 @@ TEST(Sensor_TEST, SetRateZeroService)
   EXPECT_FLOAT_EQ(20.0, sensor.UpdateRate());
 }
 
+//////////////////////////////////////////////////
+TEST(Sensor_TEST, FrameIdFromSdf)
+{
+  auto loadSensorWithSdfParam =
+      [](TestSensor &_testSensor, const std::string &_sensorParam)
+  {
+    const std::string sensorSdf = R"(
+    <sdf version="1.9">
+      <model name="m1">
+        <link name="link1">
+          <sensor name="test" type="imu">)" +
+                                  _sensorParam + R"(
+          </sensor>
+        </link>
+      </model>
+    </sdf>
+    )";
+    sdf::Root root;
+    sdf::Errors errors = root.LoadSdfString(sensorSdf);
+    ASSERT_TRUE(errors.empty()) << errors;
+    auto *model = root.Model();
+    ASSERT_NE(model, nullptr);
+    auto *link = model->LinkByIndex(0);
+    ASSERT_NE(link, nullptr);
+    auto *sensor = link->SensorByIndex(0);
+    ASSERT_NE(sensor, nullptr);
+
+    _testSensor.Load(*sensor);
+  };
+
+  {
+    TestSensor testSensor;
+    loadSensorWithSdfParam(testSensor, "");
+    EXPECT_EQ("test", testSensor.FrameId());
+  }
+  {
+    TestSensor testSensor;
+    loadSensorWithSdfParam(
+        testSensor, "<ignition_frame_id>custom_frame_id</ignition_frame_id>");
+    EXPECT_EQ("custom_frame_id", testSensor.FrameId());
+  }
+  {
+    TestSensor testSensor;
+    loadSensorWithSdfParam(testSensor,
+                           "<gz_frame_id>custom_frame_id</gz_frame_id>");
+    EXPECT_EQ("custom_frame_id", testSensor.FrameId());
+  }
+  {
+    TestSensor testSensor;
+    loadSensorWithSdfParam(testSensor, R"(
+      <ignition_frame_id>custom_frame_id</ignition_frame_id>
+      <gz_frame_id>other_custom_frame_id</gz_frame_id>)");
+    EXPECT_EQ("custom_frame_id", testSensor.FrameId());
```

Comment:
Test is failing here. If both `gz_frame_id` and `ignition_frame_id` elements are specified, `ign-sensors6` prefers `ignition_frame_id` (as implemented in #409) while `gz-sensors7` uses `gz_frame_id`. So I think we'll need to update test expectation here?

Reply:
I made the change (`custom_frame_id` -> `other_custom_frame_id`) and forced pushed [8f6bce6](https://github.com/gazebosim/gz-sensors/pull/412/commits/8f6bce661bdff65fbb82d64b07787f3ef92066b7)


Feel free to revert if this is not intended.

Reply:
Yup, that works, thanks!